yarn-project/simulator/src/public/executor.ts

```diff
@@ -71,13 +73,22 @@ async function executePublicFunctionAvm(executionContext: PublicExecutionContext
   const simulator = new AvmSimulator(context);
 
   const result = await simulator.execute();
-  const newWorldState = context.persistableState.flush();
-
   log.verbose(
     `[AVM] ${address.toString()}:${selector} returned, reverted: ${result.reverted}, reason: ${result.revertReason}.`,
   );
 
-  return await convertAvmResults(executionContext, newWorldState, result, machineState);
+  // Extract the call's results from AVM state and adapt it to a PublicExecutionResult
+  return Promise.resolve({
+    ...worldStateJournal.transitionalExecutionResult, // includes nestedExecutions
+    execution: executionContext.execution,
+    startSideEffectCounter: new Fr(startSideEffectCounter),
+    endSideEffectCounter: new Fr(worldStateJournal.trace.accessCounter),
+    returnValues: result.output,
+    unencryptedLogs: new UnencryptedFunctionL2Logs(worldStateJournal.transitionalExecutionResult.unencryptedLogs),
+    reverted: result.reverted,
+    revertReason: result.revertReason ? createSimulationError(result.revertReason) : undefined,
+    gasLeft: machineState.gasLeft,
+  });
 }
```

Comment:
This basically replaces the `convertAvmResults` function. We could optionally move this into the transitional file 🤷